### PR TITLE
Allow password change via api for CommCareUser, add end point for del…

### PR DESCRIPTION
**Pull request by mkangia**
…eting commcare user, add url to flush key records for CommCareUser

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 6 errors:

* corehq/apps/api/resources/v0_5.py, line 1 - F401 'HttpResponse' imported but unused
* corehq/apps/api/resources/v0_5.py, line 190 - E122 continuation line missing indentation or outdented
* corehq/apps/mobile_auth/models.py, line 53 - E124 closing bracket does not match visual indentation
* corehq/apps/mobile_auth/models.py, line 54 - W292 no newline at end of file
* corehq/apps/mobile_auth/tests.py, line 95 - W292 no newline at end of file
* corehq/apps/mobile_auth/views.py, line 94 - W391 blank line at end of file

File: corehq/apps/api/resources/v0_5.py [hunk 1]
```diff
@@ -32,6 +34,7 @@
 from corehq.apps.userreports.reports.view import query_dict_to_dict, \
     get_filter_values
 from corehq.apps.userreports.sql.columns import UCRExpandDatabaseSubcolumn
+from corehq.apps.users.forms import SetUserPasswordForm
```
Comment (dimagimon) [inline corehq/apps/api/resources/v0_5.py]: F401 'SetUserPasswordForm' imported but unused

File: corehq/apps/mobile_auth/tests.py [hunk 2]
```diff
@@ -1,24 +1,38 @@
 from datetime import datetime, timedelta
-from django.test import TestCase
+from django.test import TestCase, Client
```
Comment (dimagimon) [inline corehq/apps/mobile_auth/tests.py]: F401 'Client' imported but unused

File: corehq/apps/mobile_auth/tests.py [hunk 3]
```diff
@@ -1,24 +1,38 @@
...
 from casexml.apps.case.tests.util import check_xml_line_by_line
+
+from corehq.apps.mobile_auth.models import MobileAuthKeyRecord
```
Comment (dimagimon) [inline corehq/apps/mobile_auth/tests.py]: F401 'MobileAuthKeyRecord' imported but unused

File: corehq/apps/mobile_auth/tests.py [hunk 4]
```diff
@@ -1,24 +1,38 @@
...
 from corehq.apps.mobile_auth.utils import new_key_record, get_mobile_auth_payload
 from dimagi.ext.jsonobject import HISTORICAL_DATETIME_FORMAT
+from corehq.apps.domain.shortcuts import create_domain
+from django.core.urlresolvers import reverse
```
Comment (dimagimon) [inline corehq/apps/mobile_auth/tests.py]: F401 'reverse' imported but unused

**Comment by mkangia** [comment]
@snopoke Changes removed for flushing keys.
Any review points for changes done? Should we wait for further info from @phillipm or ship the current changes out(post review)?

**Comment by snopoke** [comment]
Build failing:
```
======================================================================
ERROR: test_case_list_media (corehq.apps.app_manager.tests.test_media_suite.MediaSuiteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/django/test/utils.py", line 216, in inner
    return test_func(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_media_suite.py", line 66, in test_case_list_media
    self.assertXmlEqual(self.get_xml('media_suite'), app.create_media_suite())
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/models.py", line 5208, in create_media_suite
    return MediaSuiteGenerator(self, build_profile_id).generate_suite()
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/suite_xml/generator.py", line 92, in generate_suite
    self.suite.media_resources.extend(self.media_resources)
  File "/vendor/lib/python2.7/site-packages/eulxml/xmlmap/fields.py", line 763, in extend
    for item in list:
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/suite_xml/generator.py", line 151, in media_resources
    ) + urllib.quote(name.encode('utf-8')) if name else name
  File "/vendor/lib/python2.7/site-packages/django/core/urlresolvers.py", line 551, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/vendor/lib/python2.7/site-packages/django/core/urlresolvers.py", line 415, in _reverse_with_prefix
    self._populate()
  File "/vendor/lib/python2.7/site-packages/django/core/urlresolvers.py", line 292, in _populate
    for name in pattern.reverse_dict:
  File "/vendor/lib/python2.7/site-packages/django/core/urlresolvers.py", line 315, in reverse_dict
    self._populate()
  File "/vendor/lib/python2.7/site-packages/django/core/urlresolvers.py", line 292, in _populate
    for name in pattern.reverse_dict:
  File "/vendor/lib/python2.7/site-packages/django/core/urlresolvers.py", line 315, in reverse_dict
    self._populate()
  File "/vendor/lib/python2.7/site-packages/django/core/urlresolvers.py", line 303, in _populate
    lookups.appendlist(pattern.callback, (bits, p_pattern, pattern.default_args))
  File "/vendor/lib/python2.7/site-packages/django/core/urlresolvers.py", line 231, in callback
    self._callback = get_callable(self._callback_str)
  File "/vendor/lib/python2.7/site-packages/django/utils/lru_cache.py", line 101, in wrapper
    result = user_function(*args, **kwds)
  File "/vendor/lib/python2.7/site-packages/django/core/urlresolvers.py", line 118, in get_callable
    (lookup_view, mod_name))
ViewDoesNotExist: Could not import corehq.apps.mobile_auth.views.flush_key_records. View does not exist in module corehq.apps.mobile_auth.views.
```

**Comment by mkangia** [comment]
@snopoke Thanks simon. Build fixed.

**Comment by snopoke** [comment]
:+1: looks good - still has don't merge label though

**Comment by mkangia** [comment]
@snopoke Label removed!

**Pull request merged**